src/shadowbox/server/manager_service.ts

```diff
@@ -151,12 +152,15 @@ export class ShadowsocksManagerService {
   }
 
   // Sets the default ports for new access keys
-  public async setDefaultPort(req: RequestType, res: ResponseType, next: restify.Next):
+  public async setPortForNewAccessKeys(req: RequestType, res: ResponseType, next: restify.Next):
       Promise<void> {
     try {
-      logging.debug(`setDefaultPort request ${JSON.stringify(req.params)}`);
+      logging.debug(`setPortForNewAccessKeys request ${JSON.stringify(req.params)}`);
       const port = req.params.port;
```

Comment:
Let's parse req.params.port as a number and handle possible errors.

Reply:
This is obsolete now that we changed it pack to a number

Reply:
You still need to check that the port is a number before passing to `setPortForNewAccessKeys`. I tried `Number.isNaN("foo")` and is returns false.